src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java

```diff
@@ -416,8 +418,8 @@
                 } else {
                     final ProductVariant matchingOldVariant = oldProductVariantsWithMaster.get(newProductVariantKey);
                     final List<UpdateAction<Product>> updateOrAddVariant = ofNullable(matchingOldVariant)
-                        .map(oldVariant -> collectAllVariantUpdateActions(oldProduct, oldVariant, newProductVariant,
-                            attributesMetaData, syncOptions))
+                        .map(oldVariant -> collectAllVariantUpdateActions(updateActions, oldProduct, oldVariant,
```

Comment:
Wouldn't it be better if we pass only the `SetAttributeInAllVariants` update actions? passing down the `updateActions` from this place will make additional iterations inside the inner method calls. We can even remove casting in filter operations. wdyt?

Reply:
Yes sure, good idea.